File: 06-tmcra-sdk-integrations/sdk/typescript/scripts/server-smoke.mjs
import assert from "node:assert/strict";
import { writeFile } from "node:fs/promises";
import { TMCRAClient } from "../dist/index.js";

const required = [
  "TMCRA_BASE_URL",
  "TMCRA_API_KEY",
  "TMCRA_DEFAULT_SCOPE",
  "TMCRA_SMOKE_EXPECTED_TEXT",
];
for (const name of required) {
  if (!process.env[name]?.trim()) throw new Error(`${name} is required`);
}

const client = new TMCRAClient({
  baseUrl: process.env.TMCRA_BASE_URL,
  apiKey: process.env.TMCRA_API_KEY,
  defaultTimeoutMs: 60_000,
});
const health = await client.healthz();
const readiness = await client.readyz();
const recalled = await client.recall(process.env.TMCRA_DEFAULT_SCOPE, {
  query: "What is my launch verification code?",
  evidence_mode: "auto",
  max_windows: 8,
});
assert.equal(health.status, "ok");
assert.equal(readiness.status, "ready");
assert.ok(
  recalled.prompt_evidence.content.includes(process.env.TMCRA_SMOKE_EXPECTED_TEXT),
);

const report = {
  schema_version: "tmcra.typescript-server-smoke.1",
  status: "passed",
  health: health.status,
  readiness: readiness.status,
  selected_evidence_mode: recalled.evidence_route.selected,
  recalled_expected_text: true,
};
if (process.env.TMCRA_SMOKE_REPORT) {
  await writeFile(
    process.env.TMCRA_SMOKE_REPORT,
    `${JSON.stringify(report, null, 2)}\n`,
    { mode: 0o600 },
  );
}
console.log(JSON.stringify(report));

File: 08-tmcra-mcp-server/src/tmcra_mcp/server_smoke.py
from __future__ import annotations

import asyncio
import json
import os
from pathlib import Path

from mcp.shared.memory import create_connected_server_and_client_session

from .config import MCPSettings
from .server import create_server


async def _run() -> dict[str, object]:
    expected = os.getenv("TMCRA_SMOKE_EXPECTED_TEXT", "").strip()
    if not expected:
        raise RuntimeError("TMCRA_SMOKE_EXPECTED_TEXT is required")
    server = create_server(MCPSettings.from_env())
    client = getattr(server, "_tmcra_client")
    try:
        async with create_connected_server_and_client_session(server) as session:
            listed = await session.list_tools()
            names = sorted(tool.name for tool in listed.tools)
            result = await session.call_tool(
                "tmcra_recall",
                {"query": "What is my launch verification code?"},
            )
            if result.isError:
                raise RuntimeError("MCP recall returned a protocol error")
            content = str(
                ((result.structuredContent or {}).get("prompt_evidence") or {}).get(
                    "content", ""
                )
            )
            if expected not in content:
                raise RuntimeError("MCP recall did not contain the expected evidence")
            return {
                "schema_version": "tmcra.mcp-server-smoke.1",
                "status": "passed",
                "listed_tools": names,
                "protocol_error": False,
                "recalled_expected_text": True,
            }
    finally:
        await client.aclose()


def main() -> int:
    report = asyncio.run(_run())
    report_path = os.getenv("TMCRA_SMOKE_REPORT", "").strip()
    if report_path:
        Path(report_path).write_text(
            json.dumps(report, ensure_ascii=True, indent=2, sort_keys=True) + "\n",
            encoding="utf-8",
        )
        try:
            os.chmod(report_path, 0o600)
        except OSError:
            pass
    print(json.dumps(report, ensure_ascii=True, sort_keys=True))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
